js-runtime/index.js

```diff
@@ -456,6 +459,39 @@ module.exports = {
   hashf,
   hashv,
 
+  // In Node.js these opcodes don't have to be IO opcodes, but in the Rust runtime they do, because
+  // of the multithreaded nature of the Rust runtime. Not sure if they should be "fake" async here
+  // or not.
+  dssetf:  (ns, key, val) => {
+    ds[`${ns}:${key}`] = val
+  },
+  dssetv:  (ns, key, val) => {
+    ds[`${ns}:${key}`] = val
+  },
+  dshas:   (ns, key) => ds.hasOwnProperty(`${ns}:${key}`),
+  dsdel:   (ns, key) => {
+    const fullKey = `${ns}:${key}`
+    const toDelete = ds.hasOwnProperty(fullKey)
+    if (toDelete) delete ds[fullKey]
+    return toDelete
+  },
+  dsgetf:  (ns, key) => {
+    const fullKey = `${ns}:${key}`
+    if (ds.hasOwnProperty(fullKey)) {
+      return [ true, ds[`${ns}:${key}`], ]
+    } else {
+      return [ false, 'namespace-key pair not found', ]
+    }
+  },
+  dsgetv:  (ns, key) => {
+    const fullKey = `${ns}:${key}`
+    if (ds.hasOwnProperty(fullKey)) {
+      return [ true, ds[`${ns}:${key}`], ]
+    } else {
+      return [ false, 'namespace-key pair not found', ]
+    }
```

Comment:
non-blocking nit: is there any way to dry this up easily?

Reply:
Unfortunately, no, since they're defined inline with the object creation. I can't just use the prior definition. The only way is to restructure it *all* so I define the opcodes as separate variables and then have the object definition reference these variables when constructing said object.

This would *almost double* the size of the file since most of the time the opcodes are one-liners.